superset/connectors/sqla/models.py

```diff
@@ -155,6 +159,42 @@ def dttm_sql_literal(self, dttm):
                 self.type or '', dttm)
             return s or "'{}'".format(dttm.strftime('%Y-%m-%d %H:%M:%S.%f'))
 
+    def get_metrics(self):
+        metrics = []
+        M = SqlMetric  # noqa
+        quoted = self.column_name
+        if self.sum:
```

Comment:
Should we delete the metrics that were 'unselected'?

Reply:
I didn't want to risk it as metrics may be used in charts and users may create issues. User then have to both uncheck and explicitly delete the metric.